### PR TITLE
[Enhancement] (nereids)implement showCreateDatabaseCommand in nereids

### LINKED ISSUE #42739: [Enhancement] (nereids)implement showCreateDatabaseCommand in nereids
showCreateDatabase command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showCreateDatabase command in nereids:
unsupportedshowCreateDatabaseStatement
``` SHOW CREATE (DATABASE | SCHEMA) name=multipartIdentifier                      ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42739 

Implement showCreateDatabaseCommand in nereids

Changed  Session  Variables:
VarName                                                                                |  CurrentValue                |  DefaultValue
-----------------------------------------------|---------------------|-------------
character_set_results                                                    |  NULL                                |  utf8mb4          
enable_parallel_result_sink                                        |  true                                |  false              
sql_mode                                                                              |  STRICT_TRANS_TABLES  |                          
variable_version                                                              |  0                                      |  300                  
experimental_enable_nereids_distribute_planner  |  false                              |  true                

2024-11-19 14:20:42.684 INFO [suite-thread-1] (Suite.groovy:1316) - Execute tag: cmd, sql: SHOW CREATE DATABASE db_test_show_create
2024-11-19 14:20:42.691 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: DROP DATABASE IF EXISTS db_test_show_create
2024-11-19 14:20:42.707 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_show_create_db_nereids in /root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_db_nereids.groovy succeed
2024-11-19 14:21:01.846 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-19 14:22:18.677 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_db_nereids.groovy: group=query,create_database,p0, name=test_show_create_db_nereids
2024-11-19 14:22:18.678 INFO [main] (RegressionTest.groovy:492) - All suites success.

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateDatabaseCommand.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents the command for SHOW CREATE DATABASE.
+ */
+public class ShowCreateDatabaseCommand extends Command implements NoForward {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateDatabaseCommand.java:51]:
```suggestion
public class ShowCreateDatabaseCommand extends ShowCommand {
```

**Comment by starocean999** [comment]
please add test case

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run p0

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Comment by msridhar78** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**